scripts/eessi-bot-build.slurm

```diff
@@ -25,27 +20,137 @@
+      eessi_tmpdir=$tmpdir
+      shift 2
+      ;;
+    -x|--http-proxy)
+      http_proxy="$2"
+      install_software_opts="$install_software_opts --http-proxy $http_proxy"
+      shift 2
+      ;;
+    -y|--https-proxy)
+      https_proxy="$2"
+      install_software_opts="$install_software_opts --https-proxy $https_proxy"
+      shift 2
+      ;;
+    -*|--*)
+      echo "Error: Unknown option: $1" >&2
+      exit 1
+      ;;
+    *)  # No more options
+      POSITIONAL_ARGS+=("$1") # save positional arg
+      shift
+      ;;
+  esac
+done
+
+echo "Starting eessi-bot-build-slurm.sh"
+
+echo -n "updating \$eessi_tmpdir: '$eessi_tmpdir' -> "
+# replace any env variable in $eessi_tmpdir with its
+#   current value (e.g., a value that is local to the job)
+eessi_tmpdir=$(envsubst <<< $eessi_tmpdir)
+echo "'$eessi_tmpdir'"
 
 # bind current working directory to /eessi_bot_job in build container,
 # and also use it as home directory in build container
 export SINGULARITY_HOME="$(pwd):/eessi_bot_job"
 
+# set tmp directory to deal with possibly too small /tmp
+export SINGULARITY_TMPDIR=$eessi_tmpdir/singularity_tmpdir
+mkdir -p $SINGULARITY_TMPDIR
+
+# if eessi_tmpdir is not starting with '/tmp' map it to '/tmp'
+# rationale: something else than /tmp is used for eessi_tmpdir
+#            if /tmp has some limitations (e.g., size);
+#            the bot job is then provided with a (job-specific)
+#            alternative via the bot's config key local_tmp;
+#            because many downstream scripts assume temporary
+#            storage under /tmp, the alternative is mapped to
+#            /tmp inside the container;
+#            the mapping is only needed if $eessi_tmpdir not
+#            already points to /tmp
+if [[ $eessi_tmpdir != /tmp* ]] ;
+then
+    export SINGULARITY_BIND="$eessi_tmpdir:/tmp"
+fi
+
+echo "SINGULARITY_BIND=${SINGULARITY_BIND}"
+
+# load modules if parameter --load-modules (or environment
+#   variable load_modules) was given
+if [[ ! -z $load_modules ]]; then
+  for m in $(echo $load_modules | tr ',' '\n')
+  do
+    echo "eessi-bot-build.slurm: module load '$m'"
+    module load $m
+  done
+else
+  echo "eessi-bot-build.slurm: no modules to be loaded"
+fi
+
 # run standard EESSI software-layer install script
 #   in compat layer environment inside build container
-./build_container.sh run ${eessi_tmpdir} ./install_software_layer.sh
+./build_container.sh run ${eessi_tmpdir} \
+    ./install_software_layer.sh $install_software_opts $common_opts "$@"
 
 # create tarball for the above build and place it in some specific
 #   directory on shared disk (just in main directory of job)
 #   see https://github.com/EESSI/eessi-bot-software-layer/issues/7
-EESSI_SILENT=1 source init/eessi_environment_variables
-software_subdir=$(echo ${EESSI_SOFTWARE_SUBDIR} | tr '/' '-')
-timestamp=$(date +%s)
-export TGZ=$(printf "eessi-%s-software-%s-%s-%d.tar.gz" ${EESSI_PILOT_VERSION} ${EESSI_OS_TYPE} ${software_subdir} ${timestamp})
```

Comment:
@trz42 Determining the host CPU arch can now also be done via the build container:

```
cd software-layer
singularity exec -B $PWD:$PWD docker://ghcr.io/eessi/build-node:debian10 python3 $PWD/eessi_software_subdir.py 2>&1 | tail -1
```

or via the `build_container.sh` script:

```
./build_container.sh run /mnt/$USER/EESSI python3 ./eessi_software_subdir.py 2>&1 | tail -1 | sed 's/.* //g'
```

Reply:
So, you can revert the changes here to keep using the current `create_tarball.sh` script:

```
source init/minimal_eessi_env  # for $EESSI_PILOT_VERSION and $EESSI_OS_TYPE
software_subdir=$(./build_container.sh run /mnt/$USER/EESSI python3 ./eessi_software_subdir.py 2>&1 | tail -1 | sed 's/.* //g')
```

Reply:
Nice!

Reply:
Have added `${generic_opt}` when calling the python script (or it would not generate the correct sub directory in case the target is generic).